openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -470,26 +470,26 @@ def add_user_to_cohort(cohort, username_or_email):
             validate_email(username_or_email)
 
             try:
-                user = UnregisteredLearnerCohortAssignments.objects.get(
+                assignment = UnregisteredLearnerCohortAssignments.objects.get(
                     email=username_or_email, course_id=cohort.course_id
                 )
-                user.course_user_group = cohort
-                user.save()
+                assignment.course_user_group = cohort
+                assignment.save()
             except UnregisteredLearnerCohortAssignments.DoesNotExist:
-                user = UnregisteredLearnerCohortAssignments.objects.create(
+                assignment = UnregisteredLearnerCohortAssignments.objects.create(
                     course_user_group=cohort, email=username_or_email, course_id=cohort.course_id
                 )
 
             tracker.emit(
                 "edx.cohort.email_address_preassigned",
                 {
-                    "user_email": user.email,
+                    "user_email": assignment.email,
                     "cohort_id": cohort.id,
                     "cohort_name": cohort.name,
                 }
             )
 
-            return (user, None, True)
+            return (None, None, True)
```

Comment:
Update documentation to make it clear that user will be None when preassigned is True.